orchagent/fdborch.cpp

```diff
@@ -118,85 +128,233 @@ void FdbOrch::update(sai_fdb_event_t type, const sai_fdb_entry_t* entry, sai_obj
             return;
         }
 
         // we already have such entries
-        if (m_entries.find(update.entry) != m_entries.end())
+        auto existing_entry = m_entries.find(update.entry);
+        if (existing_entry != m_entries.end())
         {
-             SWSS_LOG_INFO("FdbOrch notification: mac %s is already in bv_id 0x%" PRIx64,
-                    update.entry.mac.to_string().c_str(), entry->bv_id);
+             SWSS_LOG_INFO("FdbOrch LEARN notification: mac %s is already in bv_id 0x%lx existing-bp 0x%lx new-bp:0x%lx",
+                    update.entry.mac.to_string().c_str(), entry->bv_id, existing_entry->second.bridge_port_id, bridge_port_id);
              break;
         }
 
         update.add = true;
+        update.type = "dynamic";
         storeFdbEntryState(update);
+        update.port.m_fdb_count++;
+        m_portsOrch->setPort(update.port.m_alias, update.port);
+        vlan.m_fdb_count++;
+        m_portsOrch->setPort(vlan.m_alias, vlan);
 
         for (auto observer: m_observers)
         {
             observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
         }
 
         break;
-
+    }
     case SAI_FDB_EVENT_AGED:
-    case SAI_FDB_EVENT_MOVE:
+    {
+        SWSS_LOG_INFO("Received AGE event for bvid=%lx mac=%s port=%lx", entry->bv_id, update.entry.mac.to_string().c_str(), bridge_port_id);
+
+        if (!m_portsOrch->getPort(entry->bv_id, vlan))
+        {
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to locate vlan port from bv_id 0x%lx", entry->bv_id);
+            return;
+        }
+
+        auto existing_entry = m_entries.find(update.entry);
+        // we don't have such entries
+        if (existing_entry == m_entries.end())
+        {
+             SWSS_LOG_INFO("FdbOrch AGE notification: mac %s is not present in bv_id 0x%lx bp 0x%lx",
+                    update.entry.mac.to_string().c_str(), entry->bv_id, bridge_port_id);
+             break;
+        }
+
+        if (!m_portsOrch->getPortByBridgePortId(bridge_port_id, update.port))
+        {
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to get port by bridge port ID 0x%lx", bridge_port_id);
+            return;
+        }
+
+        if (existing_entry->second.bridge_port_id != bridge_port_id)
+        {
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Stale aging event received for mac-bv_id %s-0x%lx with bp=0x%lx existing bp=0x%lx", update.entry.mac.to_string().c_str(), entry->bv_id, bridge_port_id, existing_entry->second.bridge_port_id);
+            // We need to get the port for bridge-port in existing fdb
+            if (!m_portsOrch->getPortByBridgePortId(existing_entry->second.bridge_port_id, update.port))
+            {
+                SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to get port by bridge port ID 0x%lx", existing_entry->second.bridge_port_id);
+                return;
```

Comment:
It seems bridge_port_id doesn't need to be compared and should not return.The key to delete FDB entry are vid and MAC, no matter the port. Even though the bridge_port_id is not matched, this MAC still need to be deleted.

Reply:
Bridge port is not deleted until all fdb entries are removed from the bridge port. If bridge port is not found when AGE event is received, it's an error condition. We can go ahead and delete the fdb, but it doesn't help as its already in a bad state.